modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/store/FilePageStoreManager.java

```diff
@@ -305,14 +340,99 @@ private List<FilePageStore> createFilePageStores(
 
         try {
             for (int i = 0; i < partitions; i++) {
-                Path partFilePath = groupWorkDir.resolve(String.format(PART_FILE_TEMPLATE, i));
+                int part = i;
+
+                Path partFilePath = groupWorkDir.resolve(String.format(PART_FILE_TEMPLATE, part));
+
+                Path[] partDeltaFiles = findPartitionDeltaFiles(groupWorkDir, partitions);
+
+                FilePageStore filePageStore = filePageStoreFactory.createPageStore(buffer.rewind(), partFilePath, partDeltaFiles);
+
+                filePageStore.setDeltaFilePageStoreIoFactory((index, pageIndexes) -> createLatest(groupWorkDir, part, index, pageIndexes));
+
+                filePageStore.setCompleteCreationDeltaFilePageStoreIoCallback(deltaIo -> renameDeltaFile(groupWorkDir, part, deltaIo));
```

Comment:
Renaming the file here looks very strange. So, checkpoint will call a method on file store, that method will trigger the listener and it will call this code. Why does it have to be so complicated?

Reply:
We personally decided that I would delete it.

Reply:
fix it